pkg/agent/flowexporter/connections/conntrack_linux.go

```diff
@@ -24,121 +24,92 @@ import (
+	klog.V(2).Infof("No. of flow exporter considered flows in Antrea zoneID: %d", len(filteredConns))
 
 	return filteredConns, nil
 }
 
-// connTrackSystem implements ConnTrackInterfacer
-var _ ConnTrackInterfacer = new(connTrackSystem)
-
-// ConnTrackInterfacer is an interface created to consume the required functions from the third party
-// conntrack library. This is helpful in writing unit tests.
-type ConnTrackInterfacer interface {
+// NetFilterConnTrack interface helps for testing the code that contains the third party library functions ("github.com/ti-mo/conntrack")
+type NetFilterConnTrack interface {
 	Dial() error
-	DumpFilter(filter conntrack.Filter) ([]conntrack.Flow, error)
+	DumpFilter(filter conntrack.Filter) ([]*flowexporter.Connection, error)
 }
 
-type connTrackSystem struct {
+type netFilterConnTrack struct {
 	netlinkConn *conntrack.Conn
 }
 
-func NewConnTrackInterfacer() *connTrackSystem {
-	// Ensure net.netfilter.nf_conntrack_acct value to be 1. This will enable flow exporter to export stats of connections.
-	// Do not handle error and continue with creation of interfacer object as we can still dump flows with no stats.
-	// If log says permission error, please ensure net.netfilter.nf_conntrack_acct to be set to 1.
-	sysctl.EnsureSysctlNetValue("netfilter/nf_conntrack_acct", 1)
-	// Ensure net.netfilter.nf_conntrack_timestamp value to be 1. This will enable flow exporter to export timestamps of connections.
-	// Do not handle error and continue with creation of interfacer object as we can still dump flows with no timestamps.
-	// If log says permission error, please ensure net.netfilter.nf_conntrack_timestamp to be set to 1.
-	sysctl.EnsureSysctlNetValue("netfilter/nf_conntrack_timestamp", 1)
-
-	return &connTrackSystem{}
-}
-
-func (c *connTrackSystem) Dial() error {
-	// Get conntrack in current namespace
+func (nfct *netFilterConnTrack) Dial() error {
+	// Get netlink client in current namespace
 	conn, err := conntrack.Dial(nil)
 	if err != nil {
-		klog.Errorf("Error when dialing conntrack: %v", err)
 		return err
 	}
-	c.netlinkConn = conn
+	nfct.netlinkConn = conn
 	return nil
 }
 
-func (c *connTrackSystem) DumpFilter(filter conntrack.Filter) ([]conntrack.Flow, error) {
-	conns, err := c.netlinkConn.DumpFilter(filter)
+func (nfct *netFilterConnTrack) DumpFilter(filter conntrack.Filter) ([]*flowexporter.Connection, error) {
+	conns, err := nfct.netlinkConn.DumpFilter(filter)
 	if err != nil {
-		klog.Errorf("Error when dumping flows from conntrack: %v", err)
 		return nil, err
 	}
-	return conns, nil
+	antreaConns := make([]*flowexporter.Connection, len(conns))
+	for i, conn := range conns {
+		antreaConns[i] = createAntreaConn(&conn)
+	}
+
+	klog.V(2).Infof("Finished dumping -- total no. of flows in conntrack: %d", len(antreaConns))
+
+	nfct.netlinkConn.Close()
+	return antreaConns, nil
 }
 
 func createAntreaConn(conn *conntrack.Flow) *flowexporter.Connection {
```

Comment:
not part of this PR, but I think with our coding conventions, `newAntreaConnection` would have been a better name, no?